### PR TITLE
Update react-native-screens to 3.27.0

### PR DESCRIPTION
**Pull request by gabrieldonadel**
# Why

With AGP 8, non-transitive R classes are now enabled by default, requiring the resource file names to be fully qualified. Given that React Native 0.73 depends on AGP 8.0 we must update react-native-screens to include [this](https://github.com/software-mansion/react-native-screens/pull/1919) fix.

# How

- `et uvm -m react-native-screens -c 3.27.0`

# Test Plan

- unversioned expo go + ncl screens
- bare expo + ncl screens

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: apps/bare-expo/ios/Podfile.lock [hunk 1]
```diff
@@ -1293,7 +1287,7 @@ EXTERNAL SOURCES:
 
 SPEC CHECKSUMS:
   boost: 57d2868c099736d80fcd648bf211b4431e51a558
-  DoubleConversion: 5189b271737e1565bdce30deb4a08d647e3f5f54
+  DoubleConversion: fea03f2699887d960129cc54bba7e52542b6f953
```
Comment (Kudo) [inline apps/bare-expo/ios/Podfile.lock:1290]: unrelated change?

File: apps/bare-expo/ios/Podfile.lock [hunk 2]
```diff
@@ -1364,7 +1357,7 @@ SPEC CHECKSUMS:
   FBLazyVector: 71803c074f6325f10b5ec891c443b6bbabef0ca7
   FBReactNativeSpec: 556ded527fc4354929dd12d0936a64cd95bde744
   fmt: ff9d55029c625d3757ed641535fd4a75fedc7ce9
-  glog: 04b94705f318337d7ead9e6d17c019bd9b1f6b1b
+  glog: c5d68082e772fa1c511173d6b30a9de2c05a69a2
```
Comment (Kudo) [inline apps/bare-expo/ios/Podfile.lock:1360]: unrelated change?

File: ios/Podfile.lock [hunk 3]
```diff
@@ -2766,6 +2766,6 @@ SPEC CHECKSUMS:
   Yoga: 86fed2e4d425ee4c6eab3813ba1791101ee153c6
   ZXingObjC: 8898711ab495761b2dbbdec76d90164a6d7e14c5
 
-PODFILE CHECKSUM: 2e2091adf6eb0d6921acab51e937558fd3a0e63d
+PODFILE CHECKSUM: 548074f79b4144edeae371a061bbe3a1aa0d8909
```
Comment (Kudo) [inline ios/Podfile.lock:2769]: do you have any insight for the podfile checksum change where you didn't change the podfile?

**Review by Kudo — APPROVED** [review]
looks good generally. just some nits from podfile checksum

File: apps/bare-expo/ios/Podfile.lock (hunk 2 above)
Comment (gabrieldonadel) [reply apps/bare-expo/ios/Podfile.lock]: yeah, reverted

File: apps/bare-expo/ios/Podfile.lock (hunk 1 above)
Comment (gabrieldonadel) [reply apps/bare-expo/ios/Podfile.lock]: yeah, reverted

File: ios/Podfile.lock (hunk 3 above)
Comment (gabrieldonadel) [reply ios/Podfile.lock]: I'm not 100% sure to be honest, at first I thought this could be related to my switching between so many branches but I tried reverting this and every time I run `pod install` even after deleting the `Pods` folder the checksum output is always `548074f79b4144edeae371a061bbe3a1aa0d8909 `

Comment (Kudo) [reply ios/Podfile.lock]: well np! just a curious question. feel free to merge it and we will know after merged.

**Review by lukmccall — APPROVED** [review]
